pkg/dns/dns.go

```diff
@@ -0,0 +1,417 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dns
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"hash/fnv"
+	"net"
+	"strings"
+	"time"
+
+	etcd "github.com/coreos/etcd/client"
+	skymsg "github.com/skynetservices/skydns/msg"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/endpoints"
+	kcache "k8s.io/kubernetes/pkg/client/cache"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
```

Comment:
@ArtfulCoder can you use the generated client instead? "k8s.io/kubernetes/pkg/client/unversioned" is deprecated.

You can try to use "https://github.com/kubernetes/kubernetes/tree/master/pkg/client/clientset_generated/release_1_2", then your code will be operating on versioned objects, so that future changes to the internal objects won't affect your code.

If that's too much trouble, you can use "https://github.com/kubernetes/kubernetes/tree/master/pkg/client/clientset_generated/internalclientset", that's almost the same as what you used now.